### PR TITLE
chore(deps): auto dependencies upgrade

### PR DESCRIPTION
**Pull request by iliapolo**
Added a GitHub workflow to upgrade dependencies. 

As part of this change, I also upgraded projen and migrated all dependencies to be mastered in the package.json files (i.e no version specification in .projen files).

This will eventually be properly modeled with projen via https://github.com/awslabs/cdk8s/pull/436. But since that PR will likely take some more effort, I want to push this forward in the meantime.

Example PR: https://github.com/awslabs/cdk8s/pull/460

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/workflows/dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        run: |-
+
+          # We special-case typescript because it's not semantically versionned
+          # and should remain on the same minor.
+          # https://github.com/microsoft/TypeScript/issues/14116
+
+          # We reject projen since it may involve breaking changes and we don't
+          # want to couple them with pure dependency upgrades.
+
+          # We reject constructs because ncu doesn't update pinned dev dependencies.
```
Comment (iliapolo) [inline .github/workflows/dependencies.yml:35]: @eladb Note

File: packages/cdk8s/.projenrc.js [hunk 2]
```diff
@@ -6,40 +6,45 @@ const project = new JsiiProject({
   description: 'Cloud Development Kit for Kubernetes',
   stability: common.options.stability,
 
+  // without this, the version of 'constructs' would need to be controlled
+  // from this file, since otherwise it would create a 0.0.0 dev dependency.
```
Comment (iliapolo) [inline packages/cdk8s/.projenrc.js:10]: https://github.com/projen/projen/issues/374

File: packages/cdk8s-cli/.projenrc.js [hunk 3]
```diff
@@ -27,12 +30,6 @@ const project = new TypeScriptLibraryProject({
   ...common.options,
 });
 
-// add @types/node as a regular dependency since it's needed to during "import"
-// to compile the generated jsii code.
-project.addDependencies({
-  '@types/node': Semver.caret(project.devDependencies['@types/node'].replace(/^\^/, ''))
```
Comment (iliapolo) [inline packages/cdk8s-cli/.projenrc.js:33]: @eladb I just moved this to be a regular dependency. The dev dependency version should align just like any other - or am i missing something?

File: packages/cdk8s-cli/package.json [hunk 4]
```diff
@@ -42,16 +40,17 @@
...
+    "projen": "^0.6.12",
     "standard-version": "^9.0.0",
     "ts-jest": "^26.1.0",
     "typescript": "^3.9.5"
   },
   "peerDependencies": {},
   "dependencies": {
     "@types/node": "^10.17.0",
-    "cdk8s": "0.0.0",
+    "cdk8s": "^0.0.0",
```
Comment (eladb) [inline packages/cdk8s-cli/package.json:53]: Wondering if this should be pinned

File: packages/cdk8s-plus-17/package.json [hunk 5]
```diff
@@ -70,82 +69,8 @@
     "k8s",
     "kubernetes"
   ],
-  "engines": {
-    "node": ">= 10.17.0"
-  },
```
Comment (eladb) [inline packages/cdk8s-plus-17/package.json:75]: We should specify explicit minNodeVersion in all modules.

**Review by eladb — APPROVED** [review]
A few comments. Added do not merge

File: packages/cdk8s-cli/package.json (hunk 4 above)
Comment (iliapolo) [reply packages/cdk8s-cli/package.json:53]: Probably should be

File: packages/cdk8s-plus-17/package.json (hunk 5 above)
Comment (iliapolo) [reply packages/cdk8s-plus-17/package.json:75]: added

**Pull request merged**